storage/durability/wal.rs

```diff
@@ -90,6 +90,16 @@ impl DurabilityService for WAL {
         self.registered_types.insert(Record::RECORD_TYPE, Record::RECORD_NAME);
     }
 
+    fn unsequenced_write<Record>(&self, record: &Record) -> Result<()>
```

Comment:
All WAL writes _should_ be sequenced - we can't break this API.
Epoch-based sequencing is going to be something our WAL does by default, and isn't a behaviour we want for _some_ records - this is bleeding higher-level usage information down into the implementation of the components we use.

Reply:
After an in person discussion, we decided to do unsequenced writes.